src/Penrose/Functions.hs

```diff
@@ -1292,10 +1302,37 @@ _centerArrow arr@("Arrow", _) s1@[x1, y1] s2@[x2, y2] [o1, o2] =
         ]
   in (fromx - sx) ^ 2 + (fromy - sy) ^ 2 + (tox - ex) ^ 2 + (toy - ey) ^ 2
 
+repelPt :: Autofloat a => a -> Pt2 a -> Pt2 a -> a 
+repelPt c a b = c / (distsq a b + epsd)
+
 -- | 'repel' exert an repelling force between objects
 -- TODO: temporarily written in a generic way
 -- Note: repel's energies are quite small so the function is scaled by repelWeight before being applied
 repel :: ObjFn
+
+-- TODO: factor line & arrow together OR account for the arrowhead
```

Comment:
I know the copy-pasting is not good, but I didn't see any obvious simple way to let it fall through to the more general pattern matches (using an if statement would end the pattern match)